### ISSUE TITLE
NET-1645 Fix S2930 FN: Add support of `File.OpenRead` factory method 

### ISSUE THREAD

**Issue by sebastien-marichal** [SonarSource/sonar-dotnet#8365]
From this [community post](https://community.sonarsource.com/t/s2930-dispose-when-item-is-no-longer-needed-is-not-recognized-coming-from-factory-methods/104337).

Add [File.OpenRead](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.openread?view=net-7.0) method to the list of checked factory methods.

```cs
public static void S2930FalseNegative(string filePath)
{
    FileStream fileStream = new FileStream(filePath, FileMode.Open); // Noncompliant
    FileStream fileStream2 = File.OpenRead(filePath); // FN
}
```

S2930 RSPEC needs an update to include the new entry in the list.

**Comment by antonioaversa** [comment]
@sebastien-marichal As you point out in the community thread, `File.OpenRead` is not listed in the list of methods supported by this rule, and it's expected `File.OpenRead` not to be reported.
So, following that logic, I would not consider this issue a False Negative, but rather an Improvement of the rule, since the rule is behaving as described by the [RSPEC for 2930](https://sonarsource.github.io/rspec/#/rspec/S2930).
I would then change the title of this issue into `Improve S2930 FN: Add support of File.OpenRead factory method`.

Because it would be an Improvement, and not a False Negative I would:
- either close the repro PR
- or change the `// FN` into `// Compliant: File.OpenRead not supported`

What do you think?

**Comment by sebastien-marichal** [comment]
@antonioaversa 
When I said it was expected, I meant from a technical point of view. From a user point of view, it is expected to be reported, thus I think this is an FN.
I also considered this similar [issue](https://github.com/SonarSource/sonar-dotnet/issues/6050) which is also labeled as FN

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1645](https://sonarsource.atlassian.net/browse/NET-1645)

[NET-1645]: https://sonarsource.atlassian.net/browse/NET-1645?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
